File: ignisign-ios/Public/IgnisignApplicationConfiguration.swift
//
//  IgnisignApplicationConfiguration.swift
//  ignisign-ios
//
//  Created by Marc Nigdélian on 19/12/2023.
//

import Foundation

struct IgnisignApplicationConfiguration {
    var appId: String
    var orgId: String
}

struct IgnisignApplicationEnvSettings {
    var appId: String
    var orgId: String
    var appEnv: IgnisignApplicationEnv
    var webhookds: [IgnisignWebhookSettingsDescription]
    var appRootUrl: String
    var authorizedRedirectionUrls: [String]
    var isApiKeyGenerated: Bool
    var defaultSignatureProfileId: String
}

struct IgnisignWebhookSettingsDescription {
    var _id: String
    var _createdAt: Date
    var url: String
    var description: String
}

struct IgnisignApplicationSettings {
    var appId: String
    var orgId: String
    var defaultLanguage: String
    var customerSupportEmail: String
    var logoB64: String
    var logoDarkB64: String
    var primaryColor: IgnisignApplicationVariationColor
    var secondaryColor: IgnisignApplicationVariationColor
}

struct IgnisignApplicationVariationColor {
    var _50: String
    var _100: String
    var _200: String
    var _300: String
    var _400: String
    var _500: String
    var _600: String
    var _700: String
    var _800: String
    var _900: String
    var _A100: String
    var _A200: String
    var _A300: String
    var _A400: String
    var _A700: String
}

File: ignisign-android/ignisign/src/main/java/com/ignisign/ignisign/IgnisignApplicationConfiguration.kt
package com.ignisign.ignisign

import java.util.*

data class IgnisignApplicationConfiguration(
    val appId: String,
    val orgId: String
)

data class IgnisignApplicationEnvSettings(
    val appId: String,
    val orgId: String,
    val appEnv: IgnisignApplicationEnv,
    val webhooks: List<IgnisignWebhookSettingsDescription>,
    val appRootUrl: String? = null,
    val authorizedRedirectionUrls: List<String>,
    val isApiKeyGenerated: Boolean? = null,
    val defaultSignatureProfileId: String? = null
)

data class IgnisignWebhookSettingsDescription(
    val _id: String? = null,
    val _createdAt: Date? = null,
    val url: String,
    val description: String? = null
)

data class IgnisignApplicationSettings(
    val appId: String,
    val orgId: String,
    val defaultLanguage: IgnisignSignatureLanguages? = null,
    val customerSupportEmail: String? = null,
    val logoB64: String? = null,
    val logoDarkB64: String? = null,
    val primaryColor: IgnisignApplicationVariationColor? = null,
    val secondaryColor: IgnisignApplicationVariationColor? = null
)

data class IgnisignApplicationVariationColor(
    val _50: String,
    val _100: String,
    val _200: String,
    val _300: String,
    val _400: String,
    val _500: String,
    val _600: String,
    val _700: String,
    val _800: String,
    val _900: String,
    val A100: String,
    val A200: String,
    val A400: String,
    val A700: String
)
